ckanext/etsin/mappers/datacite.py

```diff
@@ -0,0 +1,257 @@
+
+    # Map contributor
+    package_dict['contributor'] = []
+    package_dict['curator'] = []
+    package_dict['rights_holder'] = []
+    for contributor in xml.findall('.//contributor'):
+        contributor_type = contributor.find('.//contributorType').text
+        if contributor_type in ["DataCollector", "DataCurator", "DataManager", "Editor", "Producer", "ProjectLeader", "ProjectMember", "Researcher", "ResearchGroup", "Supervisor"]:
+            metax_contributor_type = "contributor"
+        elif contributor_type == "Distributor":
+            metax_contributor_type = "publisher"
+        elif contributor_type == "ContactPerson":
+            metax_contributor_type = "curator"
+        elif contributor_type == "RightsHolder":
+            metax_contributor_type = "rights_holder"
+        else:
+            continue
+        person = _get_person(contributor)
+        if not Person:
+            continue
+        package_dict[metax_contributor_type].append(person)
+
+    # Map date
+    package_dict['provenance'] = []
+    for date in xml.findall('.//date'):
+        date_type = date.get('dateType')
+        if date_type in ['todo: find correct reference data']:  # TODO
+            package_dict['provenance'].append({
+                'temporal': {
+                    'start_date': date.text,
+                    'end_date': date.text
+                },
+                'type': {
+                    'pref_label': dateType,
+                    'identifier': 'TODO'  # TODO
+                }
+            })
+
+    # Map alternate identifier
+    package_dict['other_identifier'] = []
+    for alternate_identifier in xml.findall('.//alternateIdentifier'):
+        alternate_identifier_type = alternate_identifier.get(
+            'alternateIdentifierType')
+        if alternate_identifier_type == "URL":
+            package_dict['other_identifier'].append({
+                'notation': alternate_identifier.text,
+                'type': alternate_identifierType,
+            })
+
+    # Map related identifier
+    package_dict['related_entity'] = []
+    for related_identifier in xml.findall('.//relatedIdentifier'):
+        related_identifier_type = related_identifier.get('relatedIdentifierType')
+        if related_identifier_type == "URL":
+            relation_type = related_identifier.get('relationType')
+            package_dict['related_entity'].append({
+                "identifier": related_identifier.text,
+                "description": relation_type,
+            })
+
+    # Map version
+    version = xml.find('.//publicationYear')
+    if version is not None:
+        package_dict['version_info'] = version.text
+
+    # Map rights
+    package_dict['access_rights'] = []
+    for right in xml.findall('.//rights'):
+        rights_URI = right.get('rightsURI')
+        package_dict['access_rights'].append({
+            'description': right.text,
+            'license': {'identifier': rights_URI},
```

Comment:
Can we be sure rights_URI is something that is found in license reference data?

Reply:
No, we can't. Then again, none of the ids in the license reference data are URIs, so I'm not sure if using them would be any better.

What's the policy on checking reference data? Kielipankki refiner doesn't seem to check anything, it just assumes that certain values will always be in reference data.